src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java

```diff
@@ -41,11 +42,17 @@ public Lobby(Long id, String name, Visibility visibility) {
     }
 
     @Override
-    public byte[] generateQrCode(String code) {
-        String data = URLEncoder.encode(HANNIBAL_URL + code, StandardCharsets.UTF_8);
+    public void generateQrCode() throws RestClientException {
+        String data = URLEncoder.encode(HANNIBAL_URL+invitationCode, StandardCharsets.UTF_8);
         RestTemplate restTemplate = new RestTemplate();
+        String QR_API_URL = "https://api.qrserver.com/v1/create-qr-code";
         String url = QR_API_URL + "/?data=" + data + "&size=100x100";
-        return restTemplate.getForObject(url, byte[].class);
+        this.qrCode = restTemplate.getForObject(url, byte[].class);
+    }
+
+    @Override
+    public byte[] getQrCode(){
```

Comment:
Why don't you take the content of generateQrCode and place it in getQrCode?

Reply:
this would probably make it smaller, will change it